tests/main/views/test_frameworks.py

```diff
@@ -378,6 +395,158 @@ def test_update_framework_catches_db_errors(self, db_commit, open_example_framew
+
+    @pytest.mark.parametrize('framework_start_status, framework_new_status, expected_declaration_data',
+                             (
+                                 ('coming', 'open', {}),
+                                 ('open', 'pending', EXPECTED_FRAMEWORK_DECLARATION_SUPPLIER_INFO),
+                                 ('pending', 'pending', {}),
+                                 ('pending', 'live', {})
+                             ))
+    def test_updating_framework_to_pending_copies_supplier_information_to_declaration(self,
+                                                                                      framework_start_status,
+                                                                                      framework_new_status,
+                                                                                      expected_declaration_data):
+        self.framework.status = framework_start_status
+        db.session.add(self.framework)
+        db.session.commit()
+
+        self.client.post(
+            '/frameworks/g-cloud-10',
+            data=json.dumps({
+                'frameworks': {'status': framework_new_status},
+                'updated_by': 'example user'
+            }),
+            content_type="application/json"
+        )
+
+        self._assert_declaration_matches(expected_declaration_data)
+
+    def test_updating_framework_to_pending_uses_first_contact_information_entry(self):
+        """Digital Marketplace theoretically supports multiple contact informations per supplier. When copying contact
+        information across to fill in the supplier declaration, we want to take the details of the first (lowest id)
+        contact information associated with the account."""
+        db.session.add(ContactInformation(supplier_id=1,
+                                          contact_name='Our awesome contact',
+                                          email='2_is_the_new_1@contact.com',
+                                          address1="Twilight's Library",
+                                          city="Ponyville",
+                                          postcode="PO42"))
+        db.session.commit()
+
+        self.client.post(
+            '/frameworks/g-cloud-10',
+            data=json.dumps({
+                'frameworks': {'status': "pending"},
+                'updated_by': 'example user'
+            }),
+            content_type="application/json"
+        )
+
+        self._assert_declaration_matches(self.EXPECTED_FRAMEWORK_DECLARATION_SUPPLIER_INFO)
+
+    def test_updating_framework_includes_keys_with_none_value_for_null_supplier_account_information(self):
+        supplier = Supplier.query.filter(Supplier.supplier_id == 1).first()
+        supplier.registration_date = None
+        supplier.vat_number = None
+        supplier.registered_name = None
+
+        ci = ContactInformation.query.filter(ContactInformation.supplier_id == supplier.supplier_id).first()
```

Comment:
would it be ok to change `ci` to `contact_information`? Just to increase readability.

Reply:
Sure 👍